src/qdrant_client/points.rs

```diff
@@ -48,4 +53,319 @@ impl Qdrant {
         })
         .await
     }
+
+    pub async fn search_batch_points(
+        &self,
+        request: impl Into<SearchBatchPoints>,
+    ) -> Result<SearchBatchResponse> {
+        let request = &request.into();
+
+        self.with_points_client(|mut points_api| async move {
+            let result = points_api.search_batch(request.clone()).await?;
+            Ok(result.into_inner())
+        })
+        .await
+    }
+
+    pub async fn search_groups(
+        &self,
+        request: impl Into<SearchPointGroups>,
+    ) -> Result<SearchGroupsResponse> {
+        let request = &request.into();
+
+        self.with_points_client(|mut points_api| async move {
+            let result = points_api.search_groups(request.clone()).await?;
+            Ok(result.into_inner())
+        })
+        .await
+    }
+
+    pub async fn batch_updates(
+        &self,
+        request: impl Into<UpdateBatchPoints>,
+    ) -> Result<UpdateBatchResponse> {
+        let request = &request.into();
+
+        self.with_points_client(|mut points_api| async move {
+            Ok(points_api.update_batch(request.clone()).await?.into_inner())
+        })
+        .await
+    }
+
+    /// Update or insert points into the collection.
+    /// If points with given ID already exist, they will be overwritten.
+    /// Also this method does not split the points to insert to avoid timeouts.
+    /// Look at [`upsert_points_batch`] for that.
+    pub async fn upsert_points(
+        &self,
+        request: impl Into<UpsertPoints>,
+    ) -> Result<PointsOperationResponse> {
+        let request = &request.into();
+        self.with_points_client(|mut points_api| async move {
+            Ok(points_api.upsert(request.clone()).await?.into_inner())
+        })
+        .await
+    }
```

Comment:
I wonder if it would make more sense to take a reference instead, because we always pass a reference into the `with_points_client` function here.

```rust
fn upsert_points(&self, request: impl AsRef<UpsertPoints>) {
    let request = request.as_ref();
    // ...
}
```

But I'm afraid it may not accept all the data types we can handle now.

By the way, I'm just thinking out loud here. Maybe you have an insight on this?

Reply:
The advantage with Into<> is that we don't need to call .build() on the builders. It is also preserves consistency with functions like `delete_collection()` as this one needs to be Into<> in order to allow strings getting passed to it directly.

I agree that references would be cleaner and in some cases cheaper so I think both approaches have their advantages and disadvantages.

I'm not sure though how often users would actually use the exact same type again for a second request.